tests/test_hydraulic_power.py

```diff
@@ -238,6 +238,164 @@ def test_hydraulic_power(self):
+        from models.unit_cases_gas.source_sink.src.run_source_sink import (
+            GasProblem,
+        )
+
+        # Settings
+        base_folder = Path(run_source_sink.__file__).resolve().parent.parent
+
+        class GasProblemHydraulic(GasProblem):
+            def read(self):
+                super().read()
+
+                for d in self.energy_system_components["gas_demand"]:
+                    new_timeseries = self.get_timeseries(f"{d}.target_gas_demand").values*4e4
+                    self.set_timeseries(f"{d}.target_gas_demand", new_timeseries)
+            def energy_system_options(self):
+                options = super().energy_system_options()
+
+                self.gas_network_settings["head_loss_option"] = HeadLossOption.LINEARIZED_N_LINES_EQUALITY
+                self.gas_network_settings["n_linearization_lines"] = 3
+                self.gas_network_settings["minimum_velocity"] = 0.0
+                self.gas_network_settings["minimize_head_losses"] = True
+
+                return options
+
+        solution = run_optimization_problem(
+            GasProblemHydraulic,
+            base_folder=base_folder,
+            esdl_file_name="source_sink.esdl",
+            esdl_parser=ESDLFileParser,
+            profile_reader=ProfileReaderFromFile,
+            input_timeseries_file="timeseries.csv",
+        )
+
+        #TODO: add check on values for hydraulic power.
+        results = solution.extract_results()
+
+        pipe = "Pipe_4abc"
+        pipe_hp_in = results[f"{pipe}.GasIn.Hydraulic_power"]
+        pipe_hp_out = results[f"{pipe}.GasOut.Hydraulic_power"]
+        pipe_hp = results[f"{pipe}.Hydraulic_power"]
+
+        pipe_mass = results["Pipe_4abc.GasIn.mass_flow"]
+        pipe_vol_flow = results[f"{pipe}.GasOut.Q"]
+
+        # due to non linearity, every timestep on new linearized line, a doubled mass flow should result in more than doubled hydraulic power
+        np.testing.assert_array_less((pipe_hp[0]+1e-6) * (pipe_mass[1]/pipe_mass[0]), pipe_hp[1])
+        np.testing.assert_array_less((pipe_hp[1] + 1e-6) * (pipe_mass[2]/pipe_mass[1]), pipe_hp[2])
+
+
+        np.testing.assert_allclose(pipe_hp, pipe_hp_in-pipe_hp_out)
+        np.testing.assert_allclose(0, pipe_hp_out)
+
+        #TODO: use mass flow to get calculated hydraulic power
+
+        v_max = solution.gas_network_settings["maximum_velocity"]
+        pipe_diameter = solution.parameters(0)[f"{pipe}.diameter"]
+        pipe_wall_roughness = solution.energy_system_options()["wall_roughness"]
+
+        temperature = 20.0
+        pipe_length = solution.parameters(0)[f"{pipe}.length"]
+
+        v_inspect = results[f"{pipe}.GasOut.Q"] / solution.parameters(0)[f"{pipe}.area"]
+        v_points = [i * v_max / solution.gas_network_settings["n_linearization_lines"] for i in
+         range(solution.gas_network_settings["n_linearization_lines"] + 1)]
+
+        v_volumetric_flow = (np.asarray(v_points) * np.pi * pipe_diameter ** 2 / 4.0)
```

Comment:
to finish the test with an check on the absolute value of the hydraulic power loss over a line